Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -2098,22 +2099,41 @@ void CGContextShowGlyphsWithAdvances(CGContextRef context, const CGGlyph* glyphs
 #pragma endregion
 
 #pragma region Drawing Operations - Basic Shapes
+
+HRESULT __CGContext::ClearRect(CGRect rect) {
+    PushBeginDraw();
+    auto endDraw = wil::ScopeExit([this]() { PopEndDraw(); });
+
+    ComPtr<ID2D1Factory> factory = Factory();
+    ComPtr<ID2D1RectangleGeometry> rectangle;
+    RETURN_IF_FAILED(factory->CreateRectangleGeometry(__CGRectToD2D_F(rect), &rectangle));
+
+    // Transformed geometry
+    ComPtr<ID2D1TransformedGeometry> transformedRectangle;
+    RETURN_IF_FAILED(factory->CreateTransformedGeometry(rectangle.Get(),
+                                                        __CGAffineTransformToD2D_F(CGContextGetUserSpaceToDeviceSpaceTransform(this)),
+                                                        &transformedRectangle));
+
+    RETURN_IF_FAILED(PushGState());
+
+    RETURN_IF_FAILED(CurrentGState().IntersectClippingGeometry(transformedRectangle.Get(), kCGPathEOFill));
+
+    ComPtr<ID2D1SolidColorBrush> brush;
+    RETURN_IF_FAILED(deviceContext->CreateSolidColorBrush(D2D1::ColorF(D2D1::ColorF::Black, 0.0f), &brush));
```

Comment:
Nit: this should be in the gstate as a clear brush, we are likely creating too many brushes.